src/LineaStateBridge.sol

```diff
@@ -1,17 +1,251 @@
+import { IWorldIDIdentityManager } from "world-id-state-bridge/interfaces/IWorldIDIdentityManager.sol";
 import { Ownable2Step } from "@openzeppelin/contracts/access/Ownable2Step.sol";
+import { ICrossDomainOwnable3 } from "world-id-state-bridge/interfaces/ICrossDomainOwnable3.sol";
 
-// Optimism interface for cross domain messaging
-
-/// @title World ID State Bridge Optimism
-/// @author Worldcoin
-/// @notice Distributes new World ID Identity Manager roots to the Linea chain
-/// @dev This contract lives on Ethereum mainnet and works for Linea
+/// @title World ID State Bridge Linea
+/// @author Worldcoin & James Harrison
+/// @notice Distributes new World ID Identity Manager roots to an Linea Stack network
+/// @dev This contract lives on Ethereum mainnet and works for Linea.
 contract LineaStateBridge is Ownable2Step {
-    // solhint-disable-next-line no-empty-blocks
-    constructor() { }
+    ///////////////////////////////////////////////////////////////////
+    ///                           STORAGE                           ///
+    ///////////////////////////////////////////////////////////////////
+
+    /// @notice The address of the LinearWorldID contract on any Linea Stack chain
+    address public immutable lineaWorldIDAddress;
+
+    /// @notice address for Linea Stack chain Ethereum mainnet L1CrossDomainMessenger contract
+    address internal immutable crossDomainMessengerAddress;
+
+    /// @notice Ethereum mainnet worldID Address
+    address public immutable worldIDAddress;
+
+    /// @notice Amount of gas purchased on the Linea Stack chain for propagateRoot
+    uint32 internal _gasLimitPropagateRoot;
+
+    /// @notice Amount of gas purchased on the OLinea Stack chain for SetRootHistoryExpiry
+    uint32 internal _gasLimitSetRootHistoryExpiry;
+
+    /// @notice Amount of gas purchased on the Linea Stack chain for transferOwnershipOp
+    uint32 internal _gasLimitTransferOwnership;
+
+    /// @notice The default gas limit amount to buy on an Linea stack chain to do simple transactions
+    uint32 public constant DEFAULT_LINEA_GAS_LIMIT = 1_000_000;
+
+    ///////////////////////////////////////////////////////////////////
+    ///                            EVENTS                           ///
+    ///////////////////////////////////////////////////////////////////
+
+    /// @notice Emitted when the StateBridge gives ownership of the LineaWorldID contract
+    /// to the WorldID Identity Manager contract away
+    /// @param previousOwner The previous owner of the LineaWorldID contract
+    /// @param messageService The new owner of the LineaWorldID contract
+    /// @param remoteAddress Whether the ownership transfer is local (Linea/Linea Stack chain EOA/contract)
+    /// or an Ethereum EOA or contract
+    event OwnershipTransferredLinea(
+        address indexed previousOwner, address indexed messageService, address remoteAddress
+    );
```

Comment:
I guess it's not "OwnershipTransferred" anymore, because we talk about message service address change. Since we now have 2 moving parts: "message service" and "authorized remote sender", we may want 2 events for registering changes of those entities, and maybe separate methods to modify them